src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_bo.c

```diff
@@ -1210,14 +1210,18 @@ int xocl_gem_prime_mmap(struct drm_gem_object *obj, struct vm_area_struct *vma)
 	if (!obj->filp)
 		return -ENODEV;
 
+	/* Add the fake offset */
+	vma->vm_pgoff += drm_vma_node_start(&obj->vma_node);
+
 	ret = obj->filp->f_op->mmap(obj->filp, vma);
 	if (ret)
 		return ret;
+	drm_gem_object_get(obj);
```

Comment:
This will hold reference count. When will the reference count be released?

Reply:
This reference is cleaned up by the vm_close function which is calling drm_gem_vm_close as a callback. Inside drm_gem_vm_close they are using drm_gem_object_put_unlocked() function to clear the reference count.